examples/document.yaml

```diff
@@ -0,0 +1,43 @@
+# Workflow derived from https://github.com/r-lib/actions/tree/v2/examples
+# Need help debugging build failures? Start at https://github.com/r-lib/actions#where-to-find-help
+on:
+  push:
+    paths: ["R/**"]
+  pull_request:
+    paths: ["R/**"]
+
+name: Document
+
+jobs:
+  document:
+    runs-on: ubuntu-latest
+    env:
+      GITHUB_PAT: ${{ secrets.GITHUB_TOKEN }}
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
```

Comment:
Why do we need a deep checkout?

Reply:
We want to be able to access multiple commits instead of just the most recent commit. Otherwise, there could be situations where the workflows are not triggered but should be. For instance, consider the case where the most recent commit did not alter any files in the `R/` folder. If we set `fetch-depth` to 1, then these workflows will not be triggered, even if a previous commit did alter a file in the `R/` folder.

This is the same behavior as in the `render-rmarkdown` workflow: https://github.com/r-lib/actions/blob/b03bfcfd29726270488b596e0b3dc2bafc9cae0b/examples/render-rmarkdown.yaml#L15-L18